grafana-plugin/src/plugin/GrafanaPluginRootPage.tsx

```diff
@@ -54,9 +55,11 @@ export const GrafanaPluginRootPage = (props: AppRootProps) => {
       }}
     >
       {() => (
-        <Provider store={rootStore}>
-          <PluginSetup InitializedComponent={Root} {...props} />
-        </Provider>
+        <RenderConditionally shouldRender={isInitialized}>
+          <Provider store={rootStore}>
```

Comment:
This still "renders" within React if we use it like this, while being hidden on the UI. Could there be the case that the insides of `Provider` depend on data that we haven't initialized yet? I'm thinking this could be a cause of a potential issue

If so, it'd be better to run `RenderConditionally` with the other prop to lazy render the content

```jsx
<RenderConditionally shouldRender={...} render={() => <Provider... }/>
```